gramex/apps/ide/ide.py

```diff
@@ -0,0 +1,174 @@
+
+    pop_list.clear()
+
+    while True:
+        for item in json_data:
+            if append_under_parent(item):
+                pop_list.append(item)    # for each successfully processed dict, add it to pop list for removal
+
+        if len(pop_list) == len(json_data):
+            break
+        else:
+            app_log.debug("Entire list not processed, making another pass")
+            for pop_item in pop_list:
+                del json_data[json_data.index(pop_item)]
+
+            pop_list.clear()
+
+    # Processed All, now remove all ids and create final dict for writing to file
+    for li in config_dict:
+        root_node_key = li.pop('text')
+        recursive_pop_id('id', li)
+        processed_yaml_dict.update({root_node_key: li})
+
+    return processed_yaml_dict
+
+
+def ide_config_handler(handler, _yaml_file='gramex.yaml'):
+    global node_id, config_dict
+    parsed_config = {}
+    config_dict = []
+    node_id = 0
+
+    _yaml_file = handler.get_arg('filename')  # comment this line for testing with file
+    # if handler == "GET":                    # with file
+    if handler.request.method == "GET":    # using browser request
+        with open(_yaml_file) as fin:
+            cfg_data = yaml.safe_load(fin)
+        try:
+            recursive_parse_dict_create_json(cfg_data, "#", parsed_config)
+        except Exception as ex:
+            app_log.debug("Exception while creating json: " + format(ex))
+            return json.dumps({"Result": [{"Failure": "true"}]})
+
+        return json.dumps(config_dict)
+
+    # if handler == 'POST':                   #with file
+    #    with open(_yaml_file) as fin:
+    #        body = _yaml.safe_load(fin)
+    elif handler.request.method == 'POST':   # using browser request
+        body = json.loads(handler.request.body)  # comment this line for testing with file
+        try:
+            yaml_out = parse_json_create_dict(body)
+        except Exception as ex:
+            app_log.debug("Exception while parsing json: " + format(ex))
+            return json.dumps({"Result": [{"Failure": "true"}]})
+        if yaml_out:
+            with open(_yaml_file, 'w') as fout:
+                yaml.dump(yaml_out, fout)
+                app_log.info("Updated file: "+_yaml_file)
+                return(json.dumps({"Result": [{"Success": "true"}]}))
+        else:
+            app_log.debug("Failed to update file: "+_yaml_file)
+            return(json.dumps({"Result": [{"Failure": "true"}]}))
+    else:
+        app_log.debug("Unrecognized http request :" + handler.request.method)
+        return (json.dumps({"Result": [{"Failure": "true"}]}))
+
+
+'''
+if __name__ == "__main__":
+    #test GET
+    #ide_config_handler('GET', "C:/Users/sandeep.bhat/Desktop/Temp/gramex.yaml");
+    #test POST # please make sure you have back-up of file as it will be overwritten
+    #ide_config_handler('POST', "C:/Users/sandeep.bhat/Desktop/Temp/yaml_file2.yaml");
+'''
```

Comment:
I did not review this file, but I'm confident that it can be under 60 lines -- even with comments and logging. Could you please shorten this?

Reply:
Already dropped you a mail to connect tomorrow to discuss.